src/range_proof.rs

```diff
@@ -186,7 +187,7 @@ impl<P> RangeProof<P>
 where
     for<'p> &'p P: Mul<Scalar, Output = P>,
     for<'p> &'p P: Add<Output = P>,
-    P: CurvePointProtocol,
+    P: Copy + CurvePointProtocol + Precomputable,
```

Comment:
personally preference, I don't like `Copy` here
The scalars are massive and this can lead to bad design choices. By forcing the developer to call `clone`  each time a clone/copy is desired it becomes a conscious choice on each clone/copy. 

looking at the new code where the clones/copy happens, it now hides the fact that the code is doing an expensive operation.

Reply:
Removed in a new update.